src/TableFunctions/ITableFunctionCluster.h

```diff
@@ -24,15 +27,23 @@ class ITableFunctionCluster : public Base
 public:
     String getName() const override = 0;
 
-    static void updateStructureAndFormatArgumentsIfNeeded(ASTs & args, const String & structure_, const String & format_, const ContextPtr & context)
+    static void updateStructureAndFormatArgumentsIfNeeded(ASTFunction * table_function, const String & structure_, const String & format_, const ContextPtr & context)
     {
+        auto * expression_list = table_function->arguments->as<ASTExpressionList>();
+        ASTs args = expression_list->children;
+
         if (args.empty())
             throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected empty list of arguments for {}Cluster table function", Base::name);
 
-        ASTPtr cluster_name_arg = args.front();
-        args.erase(args.begin());
-        Base::updateStructureAndFormatArgumentsIfNeeded(args, structure_, format_, context);
-        args.insert(args.begin(), cluster_name_arg);
+        if (table_function->name == Base::name)
+            Base::updateStructureAndFormatArgumentsIfNeeded(args, structure_, format_, context);
+        else
```

Comment:
Format `{}Cluster` function name here instead, and throw `LOGICAL_ERROR` if we matched neither `Base::name` nor `"{Base::name}Cluster".`